aten/src/ATen/native/mps/operations/Randperm.mm

```diff
@@ -0,0 +1,80 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/native/mps/OperationUtils.h>
+#include <ATen/native/TensorFactories.h>
+
+namespace at {
+namespace native {
+
+Tensor& randperm_out_mps(int64_t n, c10::optional<Generator> generator, Tensor& result) {
+  if (!MPSDevice::getInstance()->macOS_13_0()) {
+    TORCH_WARN_ONCE("MPS: randperm op is supported natively starting from macOS 13.0. ",
+                    "Falling back on CPU. This may have performace implications.");
+
+    result = result.to("cpu");
+    result = at::randperm_out(result, n).to("mps");
+    return result;
+  }
+
+  using namespace mps;
+  TORCH_CHECK(n >= 0, "n must be non-negative, got", n);
+  TORCH_CHECK(!generator.has_value() || (generator.has_value() && result.device() == generator->device()), "Expected a '", result.device(), "' generator device but found '", generator->device(), "'");
+  check_supported_max_int_with_precision(n, result);
+  result.resize_({n});
+
+  auto uniform_mps = at::native::empty_mps(result.sizes(), kFloat, c10::nullopt, kMPS);
+  uniform_mps = at::uniform(uniform_mps, 0.0, 1.0, generator);
```

Comment:
I don't think if this is efficient. It calls runMPSGraph once here in at::uniform(), and another time at the end of randperm function.